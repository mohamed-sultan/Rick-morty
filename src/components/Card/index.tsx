import Image from "next/image";
import React from "react";
import { styled } from "@mui/system";
import { characterProps } from "../../interface";

const CardWrapper = styled("div")(({ theme }) => ({
  backgroundColor: theme.palette.primary?.white,
  boxShadow: `0 0.5rem 1rem ${theme.palette.primary?.shadow}`,
  borderRadius: "15px",
  padding: "1rem",
  marginBottom: "1rem",
  textAlign: "center",
  cursor: "pointer",
  transition: " all 0.3s ease",
  "&:hover": {
    transform: "scale(1.07)",
    zIndex: 1,
  },
}));
const ImageComponent = styled(Image)(({ theme }) => ({
  borderRadius: "15px",
}));
const Name = styled("div")(({ theme }) => ({
  fontSize: "1.2rem",
  marginBottom: " 0.5rem",
  color: theme.palette.primary?.black,
}));
const Status = styled("div")(({ theme }) => ({
  marginTop: "unset",
  marginBottom: " 1.5rem",
  color: theme.palette.primary?.gray,
}));
const Location = styled("div")(({ theme }) => ({
  fontStyle: "italic",
  color: theme.palette.primary?.gray,
}));

const Card = ({ character }: characterProps) => {
  return (
    <CardWrapper>
      <ImageComponent
        priority={true}
        src={character.image}
        width={300}
        height={300}
        alt="character"
      />
      <div>
        <Name>{character.name}</Name>
        <Status>
          {character.status} - {character.species}
        </Status>
        <Location>{character.location.name}</Location>
      </div>
    </CardWrapper>
  );
};

export default Card;
