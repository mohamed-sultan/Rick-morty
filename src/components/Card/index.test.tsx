import TestRenderer from "react-test-renderer";
import Card from "./index";

describe("Card snapshot testing", () => {
  it("renders correctly", () => {
    const MockedCharacter = {
      image: "/hdflix.png",
      name: "string",
      status: "string",
      species: "string",
      location: {
        name: "string",
      },
    };
    const tree = TestRenderer.create(
      <Card character={MockedCharacter} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
