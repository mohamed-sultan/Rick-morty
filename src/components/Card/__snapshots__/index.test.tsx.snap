// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card snapshot testing renders correctly 1`] = `
<div
  className="css-19i246o"
>
  <img
    alt="character"
    className="css-y3148r"
    data-nimg="1"
    decoding="async"
    height={300}
    onError={[Function]}
    onLoad={[Function]}
    src="/_next/image?url=%2Fhdflix.png&w=640&q=75"
    srcSet="/_next/image?url=%2Fhdflix.png&w=384&q=75 1x, /_next/image?url=%2Fhdflix.png&w=640&q=75 2x"
    style={
      {
        "color": "transparent",
      }
    }
    width={300}
  />
  <div>
    <div
      className="css-qs9xs7"
    >
      string
    </div>
    <div
      className="css-xu5fs5"
    >
      string
       - 
      string
    </div>
    <div
      className="css-1msjh1x"
    >
      string
    </div>
  </div>
</div>
`;
